server/apps/users/helpers.py

```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth.models import User
+from .models import UserProfile
+
+def update_user_profile_db(data, user, **change_info):
+    """Updates the user profile for the given user.
+
+    Args:
+        data (dict): the user profile data
+        user (User): the user to update the profile of
+    """
+
+    profile, created = UserProfile.objects.get_or_create(user=user)
+
+    profile.new_user = data["new_user"]
+    profile.age_bracket = data["age_bracket"]
+    profile.gender = data["gender"]
+    profile.self_identification = data["self_identification"]
+    profile.description = data["description"]
+    #profile = UserProfile(user=user)
+
+    # .save() updates if primary key exists, inserts otherwise.
+    profile.save()
+
+def get_or_create_user_profile(user):
```

Comment:
This is a bit confusing to me, why do we need to make a `get_or_create` function when we then just call the internal one? I guess it might just be a matter of naming this function differently, as I suspect it's designed to process the data rather than actually creating the user?

Reply:
Thanks! I'm hoping this can be simplified out as it's just marshalling the data, but I'll bear what you say in mind and give it a clearer name in case it ends up staying.

Reply:
This code has now been removed - it wasn't needed, as you suspected.